src/k8s/pkg/k8sd/api/cluster_join.go

```diff
@@ -37,16 +37,16 @@ func (e *Endpoints) postClusterJoin(s state.State, r *http.Request) response.Res
 
 	joinConfig := struct {
 		// We only care about this field from the entire join config.
-		containerdBaseDir string `yaml:"containerd-base-dir,omitempty"`
+		ContainerdBaseDir string `yaml:"containerd-base-dir,omitempty"`
```

Comment:
Can we drop a comment on why the uppercase is needed? (Fixing the marshall bug, just as a reminder to ourselves)

Reply:
Sure, done.